### PR TITLE
Add 1.23.0-rc0 changelog

### PR DESCRIPTION
**Pull request by lunny**
Notice: some of pull requests haven't been added to the changelog

- Which have been backported to v1.22
    -  https://github.com/go-gitea/gitea/pulls?q=sort%3Aupdated-desc+is%3Apr+milestone%3A1.23.0+is%3Amerged+label%3Abackport%2Fv1.22
- Which have skip-changelog
    - https://github.com/go-gitea/gitea/pulls?q=sort%3Aupdated-desc+is%3Apr+milestone%3A1.23.0+is%3Amerged+-label%3Abackport%2Fv1.22+label%3Askip-changelog
- Build and Testing
    -  https://github.com/go-gitea/gitea/pulls?q=sort%3Aupdated-desc+is%3Apr+milestone%3A1.23.0+is%3Amerged+label%3Abackport%2Fv1.22+label%3Atype%2Ftesting
    - https://github.com/go-gitea/gitea/pulls?q=sort%3Aupdated-desc+is%3Apr+milestone%3A1.23.0+is%3Amerged+label%3Abackport%2Fv1.22+label%3Atopic%2Fbuild
- No `type` label marked may be missed, I need to check one by one, but I believe those are not too many.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,352 @@ This changelog goes through the changes that have been made in each release
...
+  * Use camo.Always instead of camo.Allways (#32097)
+  * Remove SHA1 for support for ssh rsa signing (#31857)
+  * Use UTC as default timezone when schedule Actions cron tasks (#31742)
+  * Clear up old Actions logs (#31735)
+  * Fix OIDC introspection authentication (#31632)
+  * Rename project `board` -> `column` to make the UI less confusing (#30170)
+
+* SECURITY
+  * Include file extension checks in attachment API (#32151)
+  * Update JS and PY dependencies (#31940)
```
Comment (wxiaoguang) [inline CHANGELOG.md]: It could be beside other "update dependency"

**Review by wxiaoguang — DISMISSED** [review]
There are more than 800 commits, which are omitted? Maybe the omitted ones could be mentioned together.

**Comment by lunny** [comment]
> There are more than 800 commits, which are omitted? Maybe the omitted ones could be mentioned together.

Some of pull requests haven't been added to the changelog

- Which have been backported to v1.22
- Which have skip-changelog
- No `type` label marked may be missed, I need to check one by one, but I believe those are not too much.

Do you think should we list these pull requests in the changelog?

File: CHANGELOG.md (hunk 1 above)
Comment (lunny) [reply CHANGELOG.md]: I'm OK to keep them here or move to the refactor section because it mentioned there are some securities on these old versions.

**Comment by wxiaoguang** [comment]
> Do you think should we list these pull requests in the changelog?

Nope, I just mean "maybe mention them together by a summary", it's up to you.

**Comment by lunny** [comment]
I will mentioned them in the pull request so that commit message can contain them.

File: CHANGELOG.md (hunk 1 above)
Comment (wxiaoguang) [reply CHANGELOG.md]: IIRC, `Update JS and PY dependencies (#31940)` updates the development dependency, it doesn't affect end users.

So:

* merge the dependency update with others.
* mention that "Includes all security fixes backported to 1.22"

**Comment by wxiaoguang** [comment]
> commit message can contain them.

No idea what is that, maybe it would be noisy?

**Comment by lunny** [comment]
Not the contents themself but a link point to the pull request list.

File: CHANGELOG.md (hunk 1 above)
Comment (lunny) [reply CHANGELOG.md]: e18b97da7a4ecc82b44955fd968453307213fe6d

**Review by Zettat123 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,353 @@ This changelog goes through the changes that have been made in each release
...
 
+## [1.23.0-rc0](https://github.com/go-gitea/gitea/releases/tag/v1.23.0-rc0) - 2024-12-16
+
+* BREAKING
+  * Use camo.Always instead of camo.Allways (#32097)
+  * Remove SHA1 for support for ssh rsa signing (#31857)
+  * Use UTC as default timezone when schedule Actions cron tasks (#31742)
+  * Clear up old Actions logs (#31735)
+  * Fix OIDC introspection authentication (#31632)
+  * Rename project `board` -> `column` to make the UI less confusing (#30170)
```
Comment (wxiaoguang) [inline CHANGELOG.md]: If it is a "breaking" change, it means that users need to do something or experience something.

But I do not see ``` Rename project `board` -> `column`  ``` is really breaking. What end users should do? Or is there anything really broken?

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,353 @@ This changelog goes through the changes that have been made in each release
 without substantial changes to our git log; to see the highlights of what has
 been added to each release, please refer to the [blog](https://blog.gitea.com).
 
+## [1.23.0-rc0](https://github.com/go-gitea/gitea/releases/tag/v1.23.0-rc0) - 2024-12-16
+
+* BREAKING
+  * Use camo.Always instead of camo.Allways (#32097)
+  * Remove SHA1 for support for ssh rsa signing (#31857)
+  * Use UTC as default timezone when schedule Actions cron tasks (#31742)
+  * Clear up old Actions logs (#31735)
```
Comment (wxiaoguang) [inline CHANGELOG.md]:
```suggestion
  * Delete Actions logs older than 1 year by default (#31735)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,353 @@ This changelog goes through the changes that have been made in each release
...
 been added to each release, please refer to the [blog](https://blog.gitea.com).
 
+## [1.23.0-rc0](https://github.com/go-gitea/gitea/releases/tag/v1.23.0-rc0) - 2024-12-16
+
+* BREAKING
+  * Use camo.Always instead of camo.Allways (#32097)
+  * Remove SHA1 for support for ssh rsa signing (#31857)
+  * Use UTC as default timezone when schedule Actions cron tasks (#31742)
+  * Delete Actions logs older than 1 year by default (#31735)
+  * Fix OIDC introspection authentication (#31632)
```
Comment (wxiaoguang) [inline CHANGELOG.md]:
```suggestion
  * Make OIDC introspection authentication strictly require Client ID and secret (#31632)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,353 @@ This changelog goes through the changes that have been made in each release
 without substantial changes to our git log; to see the highlights of what has
 been added to each release, please refer to the [blog](https://blog.gitea.com).
 
+## [1.23.0-rc0](https://github.com/go-gitea/gitea/releases/tag/v1.23.0-rc0) - 2024-12-16
+
+* BREAKING
+  * Use camo.Always instead of camo.Allways (#32097)
```
Comment (wxiaoguang) [inline CHANGELOG.md]:
```suggestion
  * Rename config option `[camo].Allways` to `[camo].Always` (#32097)
```

File: CHANGELOG.md (hunk 2 above)
Comment (lunny) [reply CHANGELOG.md]: Just the concept names changed. Many words in the UI changed from `board` -> `column`.

Comment (wxiaoguang) [reply CHANGELOG.md]: I think users could understand it without "breaking"

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,353 @@ This changelog goes through the changes that have been made in each release
...
+  * Add priority to protected branch (#32286)
+  * Included tag search capabilities (#32045)
+  * Add option to filter board cards by labels and assignees (#31999)
+  * Add automatic light/dark option for the colorblind theme (#31997)
+  * Support migration from AWS CodeCommit (#31981)
+  * Introduce globallock as distributed locks (#31908 & #31813)
+  * Support compression for Actions logs & enable by default (#31761 & #32013)
+  * Add pure SSH LFS support (#31516)
+  * Add Passkey login support (#31504)
+  * Add issue comment when moving issues from one column to another of the project (#29311)
```
Comment (wxiaoguang) [inline CHANGELOG.md]: "adding a comment"t seems to be an "enhancement" but not a feature.

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,355 @@ This changelog goes through the changes that have been made in each release
...
+  * Add permission description for API to add repo collaborator (#31744)
+  * Add return type to GetRawFileOrLFS and GetRawFile (#31680)
+  * Add skip secondary authorization option for public oauth2 clients (#31454)
+  * Add tag protection via rest api #17862 (#31295)
+  * Document possible action types for the user activity feed API (#31196)
+  * Add topics for repository API (#31127)
+  * Add support for searching users by email (#30908)
+  * Add API endpoints for getting action jobs status (#26673)
+
+* Refactor
```
Comment (lunny) [inline CHANGELOG.md]: These should be uppercase.

File: CHANGELOG.md [hunk 8]
```diff
@@ -4,6 +4,355 @@ This changelog goes through the changes that have been made in each release
...
+  * Use GetDisplayName() instead of DisplayName() to generate rss feeds (#31687)
+  * Code editor theme enhancements (#31629)
+  * Update JS dependencies (#31616)
+  * Add bohde as maintainer (#31601)
+  * Add types for js globals (#31586)
+  * Add back esbuild-loader for .js files (#31585)
+  * Don't show hidden labels when filling out an issue template (#31576)
+  * Allow synchronizing user status from OAuth2 login providers (#31572)
+  * Display app name in the registration email title (#31562)
+  * Convert frontend code to typescript (#31559)
```
Comment (lunny) [inline CHANGELOG.md]: Move this to refactor

File: CHANGELOG.md [hunk 9]
```diff
@@ -4,6 +4,355 @@ This changelog goes through the changes that have been made in each release
...
+  * Add warning message in merge instructions when `AutodetectManualMerge` was not enabled (#31805)
+  * Add types to various low-level functions (#31781)
+  * Update JS dependencies (#31766)
+  * Remove unused code from models/repos/release.go (#31756)
+  * Support delete user email in admin panel (#31690)
+  * Add `username` to OIDC introspection response (#31688)
+  * Use GetDisplayName() instead of DisplayName() to generate rss feeds (#31687)
+  * Code editor theme enhancements (#31629)
+  * Update JS dependencies (#31616)
+  * Add bohde as maintainer (#31601)
```
Comment (lunny) [inline CHANGELOG.md]: Community tasks can be removed from this section.

Comment (lunny) [reply CHANGELOG.md]: b2a4c4cf64

File: CHANGELOG.md (hunk 8 above)
Comment (lunny) [reply CHANGELOG.md]: b2a4c4cf64

File: CHANGELOG.md (hunk 7 above)
Comment (lunny) [reply CHANGELOG.md]: b2a4c4cf64

File: CHANGELOG.md (hunk 6 above)
Comment (lunny) [reply CHANGELOG.md]: b2a4c4cf64

**Comment by wxiaoguang** [comment]
Backport 1.23 #32868

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**